src/org/zaproxy/zap/extension/ascan/OptionsVariantPanel.java

```diff
@@ -258,6 +264,7 @@ public void initParam(ScannerParam param) {
         this.getChkRPCCustom().setSelected((rpcEnabled & ScannerParam.RPC_CUSTOM) != 0);
 
         this.getExcludedParameterModel().setTokens(param.getExcludedParamList());
+        this.txtValueFilterExpression.setText("");

```

Comment:
- It should be initialised with the value from the `ScannerParam`.
- The undoable edits should be discarded after setting the value.

Reply:
Can you please explain what you mean with "The undoable edits should be discarded after setting the value." ?

Reply:
`txtValueFilterExpression.discardAllEdits()`.